File: new/new/Recipe.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace newP
{
    public class Recipe
    {
        public string NameRecipe { get; set; }
        public Category CategoryRecipe { get; set; }
        public int Time { get; set; }
        public int Difficulty { get; set; }
        public DateTime DateAdd { get; set; }
        public List<string> Ingredients { get; set; }
        public List<string> Preparation { get; set; }
        public string UserName { get; set; }
        public string Image { get; set; }
        public bool View { get; internal set; }
    }
}

File: myp/src/class/Recipe.ts
import { User } from './User';
import { Category } from './Category';

export class Recipe {
    constructor(public NameRecipe?:string,public CategoryRecipe?:Category,public Time?:number,
        public Difficulty?:string,public DateAdd?:Date,public Ingredients?:string[],public Preparation?:string[],public UserName?:User,public Image?:string,public View?:boolean){}
        

}